File: design-patterns/go/builder/vehicle/vehicle_producer.go
package main

type VehicleProducer struct {
}

func NewVehicleProducer() (vehicleProducer *VehicleProducer) {
	vehicleProducer = &VehicleProducer{}
	return
}

func (vehicleProducer *VehicleProducer) BuildCar(carBuilder *CarBuilder) *CarBuilder {
	carBuilder.AddWheel(4)
	carBuilder.AddEngine(1)
	carBuilder.AddDoor(4)
	carBuilder.AddSeat(4)
	carBuilder.AddInterior()
	return carBuilder
}

func (vehicleProducer *VehicleProducer) BuildPlane(planeBuilder *PlaneBuilder) *PlaneBuilder {
	planeBuilder.AddWheel(3)
	planeBuilder.AddEngine(2)
	planeBuilder.AddDoor(4)
	planeBuilder.AddSeat(120)
	planeBuilder.AddInterior()
	planeBuilder.AddWing(2)
	
	return planeBuilder
}

File: design-patterns/php/builder/vehicle/VehicleProducer.php
<?php
// VehicleProducer.php

namespace Builder\Vehicle;

class VehicleProducer {
    public function buildCar(CarBuilder $carBuilder): CarBuilder {
        $carBuilder->addWheel(4);
        $carBuilder->addEngine(1);
        $carBuilder->addDoor(4);
        $carBuilder->addSeat(4);
        $carBuilder->addInterior();
        
        return $carBuilder;
    }

    public function buildPlane(PlaneBuilder $planeBuilder): PlaneBuilder {
        $planeBuilder->addWheel(3);
        $planeBuilder->addEngine(2);
        $planeBuilder->addDoor(4);
        $planeBuilder->addSeat(120);
        $planeBuilder->addInterior();

        try {
            $planeBuilder->addWing(2);
        } catch (\Exception $e) {
            throw new \RuntimeException($e);
        }

        return $planeBuilder;
    }
}